File: app/src/main/java/com/textile/markeet/data/retrofit/ServiceApi.java
package com.textile.markeet.data.retrofit;

import com.textile.markeet.data.models.signin.SigninDataContainer;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

public interface ServiceApi {

    @GET
    Call<String> testResponse(@Url String url);

    @POST("mobile-login.php")
    Call<SigninDataContainer> signinForm(@QueryMap Map<String, String> parameters);
}

File: app/src/main/java/com/textile/markeet/data/retrofit/ServiceApi.kt
package com.textile.markeet.data.retrofit

import com.textile.markeet.data.models.categories.CategoriesDataContainer
import com.textile.markeet.data.models.signinup.SignInUpDataContainer

import retrofit2.Call
import retrofit2.http.*

interface ServiceApi {

    @GET
    fun testResponse(@Url url: String): Call<String>

    @POST("login")
    fun signinForm(@QueryMap parameters: Map<String, String>?): Call<SignInUpDataContainer>

    @POST("register")
    fun signupForm(@QueryMap parameters: Map<String, String>?): Call<SignInUpDataContainer>

    @GET("get_categories")
    fun getCategories(@Header("Authorization") auth: String): Call<CategoriesDataContainer>
}
